### ISSUE TITLE
Allow split-mode Replicator to be restarted if the target is down

### ISSUE THREAD

**Issue by bobvawter** [cockroachdb/replicator#902]
If Replicator is deployed in split-mode (i.e. separate staging and target databases), it should be possible for a Replicator binary to be restarted if the target database is down. Replicator would continue to stage data until the target database has been restored, at which point the staged data will be applied as per usual. We would likely need to cache information about the target schema, since that is used to sanity-check incoming mutations.

**Comment by bobvawter** [comment]
PR #980 adds persistence to the schemawatch package.

**Comment by ryanluu12345** [comment]
If I'm understanding this correctly: right now we require that both target and staging must be up for us to even start the replicator C2X service right? So we basically want to remove this requirement if staging is up but target is down

**Comment by bobvawter** [comment]
Correct. Matt has been working in PR #1000 to add a test rig for this.
